### PR TITLE
Add pull request template

### PR DESCRIPTION
**Pull request by altheaden**
This PR just adds a simple PR template with a few checklist questions.

### CODE REVIEW

**Comment by altheaden** [comment]
@xylar Could you please take a look at these checklist items when you get a chance? I added the "adding a new task" item based on what you mentioned in our meeting, but you might want me to add a little more information / sub-tasks.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+<!--
+Below are a few things we ask you or your reviewers to kindly check. 
+***Remove checks that are not relevant by deleting the line(s) below.***
+-->
+Checklist
+* [ ] User's Guide has been updated
+* [ ] Developer's Guide has been updated
+* [ ] API documentation in the Developer's Guide (`api.rst`) has any new or modified class, method and/or functions listed
+* [ ] Documentation has been [built locally](https://mpas-dev.github.io/MPAS-Analysis/latest/users_guide/quick_start.html#generating-documentation) and changes look as expected
+* [ ] A new analysis task has been added
```
Comment (xylar) [inline .github/pull_request_template.md:15]:
```suggestion
* [ ] If this PR adds a new analysis task, it has also been added to the documentation
```

Comment (xylar) [reply .github/pull_request_template.md:15]: Let's move this up under "User's Guide has been updated"

**Comment by xylar** [comment]
Not sure why CI is failing but #1019 completely changes it anyway.

**Comment by altheaden** [comment]
@xylar Fixed! And yeah, I am at a loss as to why CI is failing... I just merged this branch onto my fork with the changes from #1019 and the Actions CI over there passed with no problems.

**Review by xylar — APPROVED** [review]

**Comment by xylar** [comment]
Looks great!  Merging despite CI failures.

**Pull request merged**